src/handlers/store-standardised-invoices/fetch-s3-textract-data.test.ts

```diff
@@ -0,0 +1,430 @@
+    const result = await fetchS3TextractData(givenBucket, givenKey);
+    expect(result).toEqual(mockedS3JsonValue);
+  });
+
+  test("S3 Textract data fetcher with fetched line item groups not array", async () => {
+    mockedS3JsonValue[0].LineItemGroups = 1234;
+
+    let resultError;
+    try {
+      await fetchS3TextractData(givenBucket, givenKey);
+    } catch (error) {
+      resultError = error;
+    }
+
+    expect(resultError).toBeInstanceOf(Error);
+  });
+
+  test("S3 Textract data fetcher with fetched line items undefined", async () => {
+    mockedS3JsonValue[0].LineItemGroups[0].LineItems = undefined;
+    const result = await fetchS3TextractData(givenBucket, givenKey);
+    expect(result).toEqual(mockedS3JsonValue);
+  });
+
+  test("S3 Textract data fetcher with fetched line items defined but not array", async () => {
+    mockedS3JsonValue[0].LineItemGroups[0].LineItems = 1234;
+
+    let resultError;
+    try {
+      await fetchS3TextractData(givenBucket, givenKey);
+    } catch (error) {
+      resultError = error;
+    }
+
+    expect(resultError).toBeInstanceOf(Error);
+  });
+
+  test("S3 Textract data fetcher with fetched line item with fields undefined", async () => {
+    mockedS3JsonValue[0].LineItemGroups[0].LineItems[0].LineItemExpenseFields =
+      undefined;
+
+    const result = await fetchS3TextractData(givenBucket, givenKey);
+
+    expect(result).toEqual(mockedS3JsonValue);
+  });
+
+  test("S3 Textract data fetcher with fetched line item with fields defined but not array", async () => {
+    mockedS3JsonValue[0].LineItemGroups[0].LineItems[0].LineItemExpenseFields = 1234;
+
+    let resultError;
+    try {
+      await fetchS3TextractData(givenBucket, givenKey);
+    } catch (error) {
+      resultError = error;
+    }
+
+    expect(resultError).toBeInstanceOf(Error);
+  });
+
+  test("S3 Textract data fetcher with fetched line item field not object", async () => {
+    mockedS3JsonValue[0].LineItemGroups[0].LineItems[0].LineItemExpenseFields[0] = 1234;
+
+    let resultError;
+    try {
+      await fetchS3TextractData(givenBucket, givenKey);
+    } catch (error) {
+      resultError = error;
+    }
+
+    expect(resultError).toBeInstanceOf(Error);
+  });
+
+  test("S3 Textract data fetcher with fetched line item field empty object", async () => {
+    mockedS3JsonValue[0].LineItemGroups[0].LineItems[0].LineItemExpenseFields[0] =
+      {};
+
+    const result = await fetchS3TextractData(givenBucket, givenKey);
+
+    expect(result).toEqual(mockedS3JsonValue);
+  });
+
+  test("S3 Textract data fetcher with fetched line item field type defined but not object", async () => {
+    mockedS3JsonValue[0].LineItemGroups[0].LineItems[0].LineItemExpenseFields[0].Type = 1234;
+
+    let resultError;
+    try {
+      await fetchS3TextractData(givenBucket, givenKey);
+    } catch (error) {
+      resultError = error;
+    }
+
+    expect(resultError).toBeInstanceOf(Error);
```

Comment:
With my above comment you can check for keywords in your error message. Then you can easily check if the correct error message is returned, not just any error.